File: JAVA/Day5/ParkingSystemFunction.java
import java.util.ArrayList;

public class ParkingSystemFunction {
ArrayList<String> slots = new ArrayList<>();

public static ArrayList<String> carSlots(ArrayList<String> slots){
String empty = "0";
for (int index = 0; index < 20; index++){
slots.add(empty);
}
return slots;
}

public static ArrayList<String> parkCarFromRight(ArrayList<String> slots, String choice){
for (int index = 19; index >= 0; index--){
if (choice.toLowerCase() == "right" && slots.get(index) == "0"){
slots.remove(index);
slots.add(index, "1");
break;
}
}
return slots;
}

public static ArrayList<String> parkCarFromLeft(ArrayList<String> slots, String choice){

for (int index = 0; index < 20 ; index++){
if (choice.toLowerCase() == "left" && slots.get(index) == "0"){
slots.remove(index);
slots.add(index, "1");
break;
}
}
return slots;
}

public static ArrayList<String> removeCarFromLeftOrRight(ArrayList<String> slots, int choice){
for (int index = 0; index < 20 ; index++){
if (slots.get(index) == "1"){
slots.remove(choice);
slots.add(choice, "0");
break;
}
}
return slots;
}



}

File: PYTHON/Day5/ParkingSystemFunction.py
def car_slots(slots_list):
	empty = 0
	for index in range(1, 21):
		slots_list.append(empty)
	return slots_list
	
def park_car_from_right(slots_list, choice):
	for index in range(19, 1, -1):
		if choice == "right" and slots_list[index] == 0:
			slots_list[index] += 1;
			break 
		if type(choice) != type("a"):
			raise ValueError("Input must be a string")
	return slots_list

def park_car_from_left(slots_list, choice):
	for index in range(0, 20):
		if choice == "left" and slots_list[index] == 0:
			slots_list[index] += 1;
			break 
		if type(choice) != type("a"):
			raise ValueError("Input must be a string")	
	return slots_list
	
def remove_car_from_left_or_right(slots_list, choice):
	for index in range(0, 20):
		if choice == index and slots_list[index] == 1:
			slots_list[choice] -= 1;
			break
		if type(choice) != type(4):
			raise ValueError("Input must be a number")	
	 
	return slots_list
	

	





